kernel/src/paging.rs

```diff
@@ -1,12 +1,73 @@
 pub use crate::arch_api::paging::{get_physical_address, map_page, unmap_page, PAGE_SIZE};
-use crate::{arch_api::paging::MemoryType, physical_memory_manager::PAGES_PER_BLOCK};
+use crate::physical_memory_manager::PAGES_PER_BLOCK;
 
-pub fn map_block(virtual_address: usize, physical_address: usize, memory_type: MemoryType) {
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum MemoryType {
+    Normal,
+    Device,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct PagePermissions {
+    pub user: bool,
+    pub writable: bool,
+    pub executable: bool,
+}
+
+impl PagePermissions {
+    pub fn new(user: bool, writable: bool, executable: bool) -> Self {
+        Self {
+            user,
+            writable,
+            executable,
+        }
+    }
+
+    pub const READ_ONLY: Self = Self {
```

Comment:
Do these first three permissions apply only to memory used by the kernel? Maybe we should use a KERNEL_ prefix if so?

Are these permission modes arbitrary (we make them up) or are they known permission states supported by the CPU architecture?

Reply:
Technically there are other supported combinations as well (e.g. write/execute), but I thought these were probably the most common/useful ones. I'll put the KERNEL_ prefix though.